src/box/box.h

```diff
@@ -372,6 +372,17 @@ box_select(uint32_t space_id, uint32_t index_id,
 	   const char **packed_pos, const char **packed_pos_end,
 	   bool update_pos, struct port *port);
 
+/**
+ * A special wrapper for FFI - workaround for M1.
+ * Use 64-bit integers beyond the 8th argument.
+ * See https://github.com/LuaJIT/LuaJIT/issues/205 for details.
+ */
+int
+box_select_ffi(uint32_t space_id, uint32_t index_id, const char *key,
+	       const char *key_end, const char **packed_pos,
+	       const char **packed_pos_end, bool update_pos, struct port *port,
+	       int64_t iterator, uint64_t offset, uint64_t limit);
+
```

Comment:
There's no need to add a function declaration to the header if it'll only be used via Lua FFI. Please remove.

Reply:
Done
<details>
<summary> diff </summary>

```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index 09ad27010..362b3df02 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -3075,7 +3075,12 @@ fail:
        return -1;
 }

-int
+/**
+ * A special wrapper for FFI - workaround for M1.
+ * Use 64-bit integers beyond the 8th argument.
+ * See https://github.com/LuaJIT/LuaJIT/issues/205 for details.
+ */
+extern "C" int
 box_select_ffi(uint32_t space_id, uint32_t index_id, const char *key,
               const char *key_end, const char **packed_pos,
               const char **packed_pos_end, bool update_pos, struct port *port,
diff --git a/src/box/box.h b/src/box/box.h
index 8d53aaba2..3661322c2 100644
--- a/src/box/box.h
+++ b/src/box/box.h
@@ -372,17 +372,6 @@ box_select(uint32_t space_id, uint32_t index_id,
           const char **packed_pos, const char **packed_pos_end,
           bool update_pos, struct port *port);

-/**
- * A special wrapper for FFI - workaround for M1.
- * Use 64-bit integers beyond the 8th argument.
- * See https://github.com/LuaJIT/LuaJIT/issues/205 for details.
- */
-int
-box_select_ffi(uint32_t space_id, uint32_t index_id, const char *key,
-              const char *key_end, const char **packed_pos,
-              const char **packed_pos_end, bool update_pos, struct port *port,
-              int64_t iterator, uint64_t offset, uint64_t limit);
-
 /** \cond public */

 /*
 ```
 
 </details>